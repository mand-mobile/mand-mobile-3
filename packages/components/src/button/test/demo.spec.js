import Demo0 from './cases/demo0'
import Demo1 from './cases/demo1'
import Demo2 from './cases/demo2'
import Demo3 from './cases/demo3'
import Demo4 from './cases/demo4'
import {renderToString} from '@vue/server-test-utils'

describe('Button', () => {
  test(`Basic`, async () => {
    expect(await renderToString(Demo0)).toMatchSnapshot()
  })
  test(`Plain`, async () => {
    expect(await renderToString(Demo1)).toMatchSnapshot()
  })
  test(`Round`, async () => {
    expect(await renderToString(Demo2)).toMatchSnapshot()
  })
  test(`Inline`, async () => {
    expect(await renderToString(Demo3)).toMatchSnapshot()
  })
  test(`Text link`, async () => {
    expect(await renderToString(Demo4)).toMatchSnapshot()
  })
})
