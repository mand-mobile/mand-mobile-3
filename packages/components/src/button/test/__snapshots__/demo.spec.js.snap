// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Basic 1`] = `
<div class="md-example-child md-example-child-button md-example-child-button-0"><button type="button" class="md-button md-button--default md-button--active md-button--block">
    <div class="md-button_inner">
      <!---->
      <p class="md-button_content">Default</p>
    </div>
  </button> <button type="button" disabled="disabled" class="md-button md-button--default md-button--inactive md-button--block">
    <div class="md-button_inner">
      <!---->
      <p class="md-button_content">Default Inactive</p>
    </div>
  </button> <button type="button" class="md-button md-button--primary md-button--active md-button--block">
    <div class="md-button_inner">
      <!---->
      <p class="md-button_content">Primary</p>
    </div>
  </button> <button type="button" disabled="disabled" class="md-button md-button--primary md-button--inactive md-button--block">
    <div class="md-button_inner">
      <!---->
      <p class="md-button_content">Primary Inactive</p>
    </div>
  </button> <button type="button" class="md-button md-button--primary md-button--active md-button--block">
    <div class="md-button_inner">
      <p class="md-button_content">Loading</p>
    </div>
  </button> <button type="button" class="md-button md-button--warning md-button--active md-button--block">
    <div class="md-button_inner">
      <!---->
      <p class="md-button_content">Warning</p>
    </div>
  </button> <button type="button" disabled="disabled" class="md-button md-button--warning md-button--inactive md-button--block">
    <div class="md-button_inner">
      <!---->
      <p class="md-button_content">Warning Inactive</p>
    </div>
  </button> <button type="button" disabled="disabled" class="md-button md-button--disabled md-button--active md-button--block">
    <div class="md-button_inner">
      <!---->
      <p class="md-button_content">Disabled</p>
    </div>
  </button></div>
`;

exports[`Button Inline 1`] = `
<div class="md-example-child  md-example-child-button md-example-child-button-3">
  <div class="md-example-section"><button type="button" class="md-button md-button--primary md-button--active md-button--inline">
      <div class="md-button_inner"><i class="md-icon icon-font md-icon-edit edit md" style="color:;"></i>
        <p class="md-button_content">Inline Primary</p>
      </div>
    </button> <button type="button" class="md-button md-button--primary md-button--active md-button--inline md-button--plain">
      <div class="md-button_inner">
        <!---->
        <p class="md-button_content">Inline Plain</p>
      </div>
    </button></div>
  <div class="md-example-section"><button type="button" class="md-button md-button--warning md-button--active md-button--inline md-button--small">
      <div class="md-button_inner">
        <!---->
        <p class="md-button_content">Warning</p>
      </div>
    </button> <button type="button" disabled="disabled" class="md-button md-button--disabled md-button--active md-button--inline md-button--small">
      <div class="md-button_inner">
        <!---->
        <p class="md-button_content">Disabled</p>
      </div>
    </button></div>
</div>
`;

exports[`Button Plain 1`] = `
<div class="md-example-child md-example-child-button md-example-child-button-1"><button type="button" class="md-button md-button--default md-button--active md-button--block md-button--plain">
    <div class="md-button_inner">
      <!---->
      <p class="md-button_content">Default &amp; Plain</p>
    </div>
  </button> <button type="button" class="md-button md-button--primary md-button--active md-button--block md-button--plain">
    <div class="md-button_inner">
      <!---->
      <p class="md-button_content">Primary &amp; Plain</p>
    </div>
  </button> <button type="button" class="md-button md-button--warning md-button--active md-button--block md-button--plain">
    <div class="md-button_inner">
      <!---->
      <p class="md-button_content">Warning &amp; Plain</p>
    </div>
  </button> <button type="button" disabled="disabled" class="md-button md-button--disabled md-button--active md-button--block md-button--plain">
    <div class="md-button_inner">
      <!---->
      <p class="md-button_content">Disabled &amp; Plain</p>
    </div>
  </button> <button type="button" class="md-button md-button--primary md-button--active md-button--block md-button--round md-button--plain">
    <div class="md-button_inner">
      <!---->
      <p class="md-button_content">Primary &amp; Plain &amp; Round</p>
    </div>
  </button> <button type="button" class="md-button md-button--primary md-button--active md-button--block md-button--round md-button--plain">
    <div class="md-button_inner">
      <p class="md-button_content">Primary &amp; Plain &amp; Loading</p>
    </div>
  </button></div>
`;

exports[`Button Round 1`] = `
<div class="md-example-child md-example-child-button md-example-child-button-2"><button type="button" class="md-button md-button--default md-button--active md-button--block md-button--round">
    <div class="md-button_inner">
      <!---->
      <p class="md-button_content">Default &amp; Round</p>
    </div>
  </button> <button type="button" class="md-button md-button--primary md-button--active md-button--block md-button--round">
    <div class="md-button_inner">
      <!---->
      <p class="md-button_content">Primary &amp; Round</p>
    </div>
  </button> <button type="button" class="md-button md-button--warning md-button--active md-button--block md-button--round">
    <div class="md-button_inner">
      <!---->
      <p class="md-button_content">Warning &amp; Round</p>
    </div>
  </button> <button type="button" disabled="disabled" class="md-button md-button--disabled md-button--active md-button--block md-button--round">
    <div class="md-button_inner">
      <!---->
      <p class="md-button_content">Disabled &amp; Round</p>
    </div>
  </button></div>
`;

exports[`Button Text link 1`] = `
<div class="md-example-child  md-example-child-button md-example-child-button-3"><button type="button" class="md-button md-button--link md-button--active md-button--block">
    <div class="md-button_inner"><i class="md-icon icon-font md-icon-message message md" style="color:;"></i>
      <p class="md-button_content">评论</p>
    </div>
  </button> <button type="button" disabled="disabled" class="md-button md-button--link md-button--inactive md-button--block">
    <div class="md-button_inner">
      <!---->
      <p class="md-button_content">去看看<i class="md-icon icon-font md-icon-arrow-right arrow-right md" style="color:;"></i></p>
    </div>
  </button></div>
`;
